docs/source/best-practices.rst

```diff
@@ -0,0 +1,214 @@
+Best Practices
+==============
+
+It is easy to get started with Dask arrays, but using them *well* requires
+some experience.  This page contains suggestions for best practices, and
+includes solutions to common problems.
+
+This document specifically focuses on best practices that are shared among all
+of the Dask APIs.  Readers may first want to investigate one of the
+API-specific Best Practices documents first.
+
+-  :doc:`Arrays <array-best-practices>`
+-  :doc:`DataFrames <dataframe-best-practices>`
+-  :doc:`Delayed <delayed-best-practices>`
+
+
+Start Small
+-----------
+
+Parallelism brings extra complexity and overhead.
+Sometimes it's necessary for larger problems, but often it's not.
+Before adding a parallel computing system like Dask to your workload you may
+want to first try some alternatives:
+
+-   **Use better algorithms or data structures**:  NumPy, Pandas, Scikit-Learn
+    may have faster functions for what you're trying to do.  It may be worth
+    consulting with an expert or reading through their docs again to find a
+    better pre-built algorithm.
+
+-   **Better file formats**:  Efficient binary formats that support random
+    access can often help you manage larger-than-memory datasets efficiently and
+    simply.
+
+-   **Compiled code**:  Compiling your Python code with Numba or Cython might
+    make parallelism unnecessary.  Or you might use the multi-core parallelism
+    available within those libraries.
+
+-   **Sampling**:  Even if you have a lot of data, there might not be much
+    advantage from using all of it.  By sampling intelligently you might be able
+    to derive the same insight from a much more manageable subset.
+
+-   **Profile**:  If you're trying to speed up slow code it's important that
+    you first understand why it is slow.  Modest time investments in profiling
+    your code can help you to identify what is slowing you down.  This
+    information can help you make better decisions about if parallelism is likely
+    to help, or if other approaches are likely to be more effective.
+
+
+Use The Dashboard
+-----------------
+
+Dask's dashboard helps you to understand the state of your workers.
+This information can help to guide you to efficient solutions.
+In parallel and distributed computing there are new costs to be aware of and so
+your old intuition may no longer be true.  Working with the dashboard can help
+you relearn about what is fast and slow and how to deal with it.
+
+See :doc:`Documentation on Dask's dashboard <diagnostics-distributed>` for more
+information.
+
+
+Avoid Very Large Graphs
+-----------------------
```

Comment:
cc @rabernat I think that this includes most of what you said earlier.  Please let me know if I'm missing things.